### ISSUE TITLE
Cannot access Trakt plugin in Plex

### ISSUE THREAD

**Issue by muskingo** [trakt/Plex-Trakt-Scrobbler#533]
Hi,

Just to clarify, I am able to access the plug-in settings, but not able to click on the plugin itself (to access the Sync menu). 

This is probably because of a recent [change](https://forums.plex.tv/t/plex-web/20528/283) to the Plex interface and the discontinuation of plugins.

Is there a workaround around this?

Thanks.

**Comment by Webdoc-01** [comment]
I too am having the same problem with this plug-in, not working in the new plex for windows, edge, chrome, or firefox, however it was working in the old plex media player.

**Comment by muskingo** [comment]
@Webdoc-01 
It's still accessible if you go to [PLEX_IP]:32400/web, but Plex devs have said that they intend to completely discontinue plugins, so this is temporary.

**Comment by Webdoc-01** [comment]
I think trakt or something like it should be a feature in plex itself.

**Comment by Webdoc-01** [comment]
Just tried your above solution, this also now looks like it no longer works, be a problem if I ever need to reinstall my system, would be unable to get data back from trakt

**Comment by muskingo** [comment]
@Webdoc-01 
Are you sure about that?

Seems to be working fine for me. 

<img width="234" alt="image" src="https://user-images.githubusercontent.com/19761269/63175338-6af03a80-c061-11e9-9ff3-009778fa0a9c.png">

**Comment by Webdoc-01** [comment]
using local IP :32400/web in edge is not working would need to load another browser to try, done a fresh windows install not got that far yet.

**Comment by Webdoc-01** [comment]
Latest google chrome 76.0.3809.100 (Official Build) (64-bit) also not working

**Comment by muskingo** [comment]
@Webdoc-01 By local IP, you do mean the local IP address of your Plex Server, right?

**Comment by Webdoc-01** [comment]
the machine the plex server is on yes

**Comment by muskingo** [comment]
@Webdoc-01 That's strange. Have you changed the port it runs on, by any chance?

**Comment by Webdoc-01** [comment]
not that I know of but will check

**Comment by muskingo** [comment]
@Webdoc-01 Are you able to access it on the machine itself (`localhost:32400/web`)?

**Comment by Webdoc-01** [comment]
Port ok page loads, but cannot click on trakt to open, I have recently updated the plex server to the latest maybe that is why

**Comment by Webdoc-01** [comment]
using machine plex server is on, is still a no go

**Comment by muskingo** [comment]
@Webdoc-01 Hmm... you're on version 1.16.4.1469? Are you sure that you're logging in directly via the IP address and not via Plex Web (`app.plex.tv`)?
What version does it show under Settings -> General?

**Comment by Webdoc-01** [comment]
I have this 1.16.5.1488 like I said above I have recently updated

**Comment by muskingo** [comment]
@Webdoc-01 it seems like you're on the Plex Beta.

**Comment by Webdoc-01** [comment]
any updates I get come on the plex pass channel

**Comment by muskingo** [comment]
Yeah that is the beta, cuz version 1.16.4.1469 is the latest on the stable channel.

**Comment by Webdoc-01** [comment]
It was only working for me in the old plex media player, and no browsers before I updated the server, now I cannot get the old media player, so I would be stuck now

**Comment by muskingo** [comment]
@Webdoc-01 You could try changing the channel to Stable and see if that improves anything.

**Comment by Webdoc-01** [comment]
Have just found a copy of plex media player (not the new plex for windows) I can click trakt and it works in that, will have to use that until a solution is found.

**Comment by Webdoc-01** [comment]
looking at blog found this

At the same time, we recognize there are some exceptions, so we will keep updating Plex Media Player until Jan 30, 2020 (you can find the downloads here).

**Comment by nemchik** [comment]
FWIW the Android app still has the ability to load the plugins interface (albeing kind of clumsily)

![image](https://user-images.githubusercontent.com/725456/63180196-5b4a0500-c013-11e9-89b6-1d0150671f04.png)
![image](https://user-images.githubusercontent.com/725456/63180263-83d1ff00-c013-11e9-86a9-fc5672e052e5.png)
![image](https://user-images.githubusercontent.com/725456/63180288-92b8b180-c013-11e9-9f18-d819cd5cb98e.png)

I imagine this will be removed eventually.

**Comment by muskingo** [comment]
After updating the iOS and the Android apps to 6.0, it appears that this is no longer possible.

**Comment by Snowburden** [comment]
From what it seems, if you put the settings from the plugin to: "Sync - Run after library updates" checked, and make sure everything is "Synchronize (Pull + Push)" you will be able to update everything.

Now it seems as though Trakt has released a new type of deal with PLEX called Plex Scrobbler, and you do it through having Trakt VIP: [https://trakt.tv/settings/scrobblers](Click Here) . BUT there is a big downfall: If you start with a clean installation of PLEX (like I did today) and you wanted to get all of your watches and unwatches synced, for some reason it does now sync your watches. But will sync what you are watching and what you just finished. 

But do it this way, and hopefully in the future the Trakt/ Plex Scrobbler will be better (the missing features from above that I mentioned.).

**Comment by rectifyer** [comment]
Yeah, we released a webhook based scrobbler knowing that Plex is phasing out their plugin support. It seems like they are moving closer and closer to fully dropping plugins unfortunately 😦. Our hope is they expand webhook support to be 2 way so we can send data to Plex to backfill your history and sync your plays from other Trakt enabled apps and media centers. 

It would also be awesome if webhooks could send individual collection data (including metadata) to sync with your Trakt collection. They send a `library.new` webhook but in my testing that only is telling us your collection was updated, but the event isn't sent for each individual item that was added.

The reason this is a Trakt VIP feature is to cover the additional infrastructure needed to listen for and process the webhooks.

**Comment by Snowburden** [comment]
It seems through some experimenting the above statement that I stated about refreshing your library, the Sync and Pull only works for Movies, and TV shows or different type of Movies do not update. Man, real shame :(

**Comment by muskingo** [comment]
There are years old threads on Plex Forums requesting full Trakt integration and that hasn’t happened yet.

**Comment by preacher65** [comment]
> It seems through some experimenting the above statement that I stated about refreshing your library, the Sync and Pull only works for Movies, and TV shows or different type of Movies do not update. Man, real shame :(

I was in a similar situation to that posted above, a clean PMS install and wanting to only carry over watched status. After reading this thread I was able to sync watched status for TV shows by using the Trakt plugin from PMP 2.40.0 - as far as I know this is the latest...? Seems to have worked. https://forums.plex.tv/t/plex-media-player/120475/85

I notice from the [Plex blog post](https://www.plex.tv/blog/desktop-af/) that the section on discontinuing PMP has been struck through and it looks as if they're continuing PMP support for now.

**Comment by muskingo** [comment]
@FormBurden what version is your PMS?

**Comment by preacher65** [comment]
Good point - 3.108.2 as I hadn't switched to the beta channel yet. So possibly that oversight was why I was able to sync?

**Comment by FormBurden** [comment]
> 
> 
> @FormBurden what version is your PMS?

It is Version 3.108.2

**Comment by muskingo** [comment]
@preacher65 @FormBurden Where did you get this number from?

**Comment by FormBurden** [comment]
> 
> 
> @preacher65 @FormBurden Where did you get this number from?

Under the settings tab when you open Plex Media Server. Straight up at the top.

**Comment by muskingo** [comment]
> (Edit; Sorry replied to a question aimed at @FormBurden but hope the info is useful.)

Yeah, no my fault for tagging him, instead of you.

**Comment by preacher65** [comment]
Sorry, too many browser windows open. That's the web client version, PMS is 1.16.5.1554

**Comment by muskingo** [comment]
It does indeed appear to be working with PMP. I'm on 1.16.5.1488 (Stable).

**Comment by FormBurden** [comment]
Oh snap, so aparently what @preacher65 posted about downloading the Plex Media Player, it still works. So if you want to update everything, go through that instead.

**Comment by muskingo** [comment]
> From what it seems, if you put the settings from the plugin to: "Sync - Run after library updates" checked, and make sure everything is "Synchronize (Pull + Push)" you will be able to update everything.

Does this work?

**Comment by FormBurden** [comment]
> 
> 
> > From what it seems, if you put the settings from the plugin to: "Sync - Run after library updates" checked, and make sure everything is "Synchronize (Pull + Push)" you will be able to update everything.
> 
> Does this work?

It only works for the Movies (original), if you have another Movies thing (like I do, Anime Movies), it does not work. This method only works for movies and nothing else for the currently new Trakt Plex Scrobbler. It is in Beta after all, hopefully it'll be improved in the future.

So from this link: https://forums.plex.tv/t/plex-media-player/120475/85

I would save that download somewhere in case you're like me and end up re-installing windows in the future. So you can get Plex good to go on your watch history and such for everything on Trakt. Use that download and save it, I would only use that player to update (that's what I'm doing) since everything works. Probably best to not update it either.

**Comment by preacher65** [comment]
1.16.5.1554 is apparently latest PMS beta, so that must have been what I downloaded and installed, as the server update channel is still set to public in Settings.

I have the Trakt settings as described above, although originally I'd just put them as pull, as I wasn't sure how well the conflict resolution would work since the Plex updates.

I don't think anything synced on library updates, and I had to use the manual "Pull from Trakt.tv" option to update, but that could just have been lag while it was pulling down new metadata. I was more concerned with getting it to sync than working out the behaviour :) so wasn't particularly scientific about not making multiple changes at once.

Relieved it's working though!

**Comment by Webdoc-01** [comment]
Don't know if this has already been posted, the other day I had to reload windows on my server meaning I had lost my data, I already knew that the old plex media player would work to get my data back.

So as I had a virgin install of plex server I took the opportunity to move my plex media server data file to another NTFS NON POOLED DRIVE, (if the data folder is placed on a pooled drive as I found out, no images show) if I have to install windows again I should have no need to pull any data back from trakt.

**Comment by muskingo** [comment]
We can only assume that in future builds of PMS, they will remove the ability to browse plugins from PMP.

Definite disappointment from Plex by not adding full trakt integration and removing plugins.

**Comment by muskingo** [comment]
At this point, I just wish Trakt could launch a standalone app that would sync the databases in the background.

**Comment by FormBurden** [comment]
Yeah I feel you. Like I said, just keep that download somewhere safe so when you have to re-download or whatever windows, or even just update your Trakt for Plex, you can do it through that PMP. Who knows when they'll update PMP to get rid of the plugin-support. So keep that PMP somewhere safe, just in case they sadly never add full integration for Trakt to Plex through their new Scrobbler.

**Comment by nemchik** [comment]
A standalone app would be great actually. I've got another app that interacts directly with the Plex database https://github.com/l3uddz/plex_autoscan to add new media differently than Plex's built in scanner, so a trakt app to sync watched status would be amazing.

**Comment by scottrobertson** [comment]
The Plex API is pretty good, so building an app on top of it may be possible. I may have a look at building something if Trakt don't get to it themselves.

**Comment by muskingo** [comment]
... For there is hope

**Comment by rectifyer** [comment]
> The Plex API is pretty good, so building an app on top of it may be possible. I may have a look at building something if Trakt don't get to it themselves.

We don't have any plans to work on anything standalone, so if you're interested please do 😄 We of course are happy to help and answer any questions, we just don't have any availability to code it ourselves right now.

**Comment by Anahkiasen** [comment]
So if I understand, the plugin does not work as Plugins are being phased out in favor of webhooks. But to use the official Trakt webhook I need *both* a Plex Pass and a Trakt VIP subscriptions? 😮

**Comment by muskingo** [comment]
@Anahkiasen yes, and even after that, you won’t be able to sync watched status with your Plex library 😕😕

**Comment by Webdoc-01** [comment]
Yes you are correct subs needed.

**Comment by Webdoc-01** [comment]
However plugin does still work but you need the old plex media player to be able to access its sync panel

**Comment by muskingo** [comment]
Not old, the current stable version of Plex Media Player is able to access plugins.

**Comment by Webdoc-01** [comment]
It can access the plugin but you cannot click on it to open sync in the current version unless it's been fixed

**Comment by Anahkiasen** [comment]
But I also checked "Synchronize after library update" and it does not synchronize at the moment, submenu or not, should it?

**Comment by muskingo** [comment]
It does for me. Watched 13 Reasons Why on Netflix and marked as watched with a trakt app, showed up within the hour.

**Comment by brw** [comment]
Plugin still works, UI doesn't, like with every plugin currently. Check out [Kitana](https://github.com/pannal/Kitana) in order to get plugin UIs back after Plex discontinued them.

**Comment by muskingo** [comment]
@brw How do I set it up on my Raspberry Pi?

**Comment by brw** [comment]
@muskingo There's installation instructions in the README (Docker and manual). It's not up to me to explain to you how to install software.

**Comment by FormBurden** [comment]
> 
> 
> It does for me. Watched 13 Reasons Why on Netflix and marked as watched with a third-party trakt app, showed up within the hour.

I was wondering what you use to sync your Netflix watches? I use a JavaScript type bookmark. Trackt.tv Sync. It's been so long since I've had it idk who's the creator of it anymore. But that's what I use to this day. Trying to see if there are better third-party app syncs for Netflix than that one. Unless that's the one you use haha?

**Comment by agneevX** [comment]
@FormBurden I use a third-party app on my phone that’s still in beta.

Since it’s done manually, I only mark watched episodes when I open it.

You can try using another app that I use for Movies called Watcht if that’s how you wanna track it.

**Comment by timothevs** [comment]
Might I suggest using OpenPHT to access Plex, and kick off the TraktTV plugin from the Channels menu? The app hasn't been updated since 2017, but works well on macOS Catalina, and Windows 10. The interface is how Plex used to be once upon a time.

**Comment by CelticWebSolutions** [comment]
> Might I suggest using OpenPHT to access Plex, and kick off the TraktTV plugin from the Channels menu? The app hasn't been updated since 2017, but works well on macOS Catalina, and Windows 10. The interface is how Plex used to be once upon a time.

Thank you for suggesting this, I'd updated my files names, then updated my library in Plex, this then made Plex put all the new files as new movies.  Causing all the watch history to be lost.  

Using OpenPHT on my Mac as you suggested allowed me to click Sync in the Trakt plugin, which I'm happy to report put all my watched history back :)  

Thank you again for the suggestion!

**Comment by agneevX** [comment]
With Plex Media Player 2.51.0, Plex has officially removed the ability to access plugins and moved to the new crappy UI. 

[You need 2.50.0 or before to access it](https://forums.plex.tv/t/plex-media-player/120475/95).

**Comment by brw** [comment]
Kitana is basically the only way forward at this point when it comes to plugin UIs. It's also made to be able to be used by multiple people (read more about that [here](https://github.com/pannal/Kitana#what-is-kitana)).
I run an instance on https://kitana.bas.sh which I'm sharing with loads of people so I thought I'd share it here too in case anyone wants to give it a try.

**Comment by Anahkiasen** [comment]
This issue is honestly the saddest issue. We had two good products and a feature that worked well for everybody, and now everything is behind a subscription gate, support for plugins was dropped like a hot ball and even if you DO pay for both subscriptions the quality of the service isn't even as good as before. This is really a shitshow on Trakt and Plex's end, particularly on Plex's. :/

**Comment by CelticWebSolutions** [comment]
> Kitana is basically the only way forward at this point when it comes to plugin UIs. It's also made to be able to be used by multiple people (read more about that [here](https://github.com/pannal/Kitana#what-is-kitana)).
> I run an instance on https://kitana.bas.sh which I'm sharing with loads of people so I thought I'd share it here too in case anyone wants to give it a try.

Hi!

Thanks for this tip, I tried yours last night and it worked, so I decided to setup my own in a docker container on my NAS. It works locally for me but I can't access it remotely.  Did you do anything to get it accessible externally?

**Comment by brw** [comment]
> > Kitana is basically the only way forward at this point when it comes to plugin UIs. It's also made to be able to be used by multiple people (read more about that [here](https://github.com/pannal/Kitana#what-is-kitana)).
> > I run an instance on [kitana.bas.sh](https://kitana.bas.sh) which I'm sharing with loads of people so I thought I'd share it here too in case anyone wants to give it a try.
> 
> Hi!
> 
> Thanks for this tip, I tried yours last night and it worked, so I decided to setup my own in a docker container on my NAS. It works locally for me but I can't access it remotely. Did you do anything to get it accessible externally?

@CelticWebSolutions If you used the standalone Docker command in the README of Kitana it should work, as long as its port is bound to 0.0.0.0/any public network interface. Perhaps your NAS has some other firewall set up or something?

I run it behind a reverse proxy with Traefik so I don't have to worry too much about accessing a container directly.
